index.js

```diff
@@ -0,0 +1,172 @@
+	async options(fieldId, search) {
+		const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		if (airports) {
+			logger.info("Sending preloaded airports");
+			return this.getIATA(airports, search);
+		} else if (fs.existsSync('./airports.json')) {
+			logger.info('Loading places from a file.');
+			const airports = require('./airports.json');
+			return this.getIATA(airports, search);
+		}
+		// } else {
+		// 	logger.info("Retrieving airports via API...");
+		// 	return request.get({
+		// 		url: `${API_BASE_URL}/autosuggest/v1.0/US/${this.config.currency}/en-US/?query=${this.config.originPlace}`,
+		// 		headers: {
+		// 			'x-rapidapi-host': 'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
+		// 			'x-rapidapi-key': this.authorization.apiKey
+		// 		},
+		// 		json: true
+		// 	}).then(body => {
+		// 		airports = body;
+		// 		return this.getIATA(airports, search);
+		// 	}).catch((error) => {
+		// 		logger.error("Caught error:", error);
+		// 	})
+		// }
+	}
+ 	/**
+   * Process a airports JSON to an options list
+   * @param {*} airports
+   * @param {String} search 
+   */
+	async getIATA(airports, search) {
+		if (search != null) {
+		  search = search.trim().toLowerCase();
+		}
+		let option = [];
+		for (const airport of airports){
+			const key = airport.iata;
+			let value = airport.name || '';
+			if (!search || value.toLowerCase().includes(search)) {
+				option.push({
+					key: key,
+					value: value
+				});
+			}	
+		}
+		return option;
+	}	
+
+	// Store price obtained from last update
+	setLastPrice(price) {
+		this.store.put("lastPrice", price);
+		return true;
+	}
+
+	// Retrieve stored price data from last update
+	getLastPrice() {
+		return this.store.get("lastPrice");
+	}
+
+	async run() {
+		// Compare the current price to the old price
+		return this.getPrice().then(new_price => {
+			const old_price = this.getLastPrice();
+			logger.info(`The new price is ${new_price}`);
+			logger.info(`The old price is ${old_price}`);
+			let color;
+			let message;
+
+			// If there is no price stored in localStorage
+			if (new_price == null) {
+				color = '#DF0101'; // red
+				message = `This flight is not listed. Please modify your request.`;
+			}
+			else if (old_price == null) {
+				color = '#088A08'; // green
+				message = `The best price for this flight is ${new_price} ${this.config.currency}.`;
+			} else if (new_price <= old_price) {
+				color = '#088A08'; // green
+				message = `The best price for this flight is ${new_price} ${this.config.currency}.
+				Let's buy it!`;
+			} else {
+				color = '#DF0101'; // red
+				message = `The best price was ${old_price} ${this.config.currency} 
+				and is now ${new_price} ${this.config.currency}`;
+			}
+			const a = new q.Signal({
+				points: [
+					[new q.Point(color)]
+				],
+				name: 'FlightPriceWatcher',
+				message: message,
+				errors: 'Error signal'
+			});
+			this.setLastPrice(new_price);
```

Comment:
I had a discussion with @supercobra , you need to change the logic.
In you situation this can happen:
- I come the morning. Price is 100
- I go to meetings all day, price goes down to 10
- At some point it goes up to 11
I come back I think the price went up when it didn't

Advice. You need to look at the trend. You can store the evolution during the day.